libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/route/ReplayRouteMapper.kt

```diff
@@ -38,9 +50,7 @@ class ReplayRouteMapper @JvmOverloads constructor(
         val geometries = directionsRoute.routeOptions()?.geometries()
         val usesPolyline6 = geometries?.contains(DirectionsCriteria.GEOMETRY_POLYLINE6) ?: false
         if (!usesPolyline6) {
-            throw IllegalStateException(
-                "Add .geometries(DirectionsCriteria.GEOMETRY_POLYLINE6) to your directions request"
-            )
+            logger?.w(tag, Message("Make sure that the route's geometry is encoded with polyline6'"))
```

Comment:
nit: this crashes with an unintuitive message. this change makes it so you have to read the logs

Reply:
Any ideas on how to handle this better? We shouldn't crash here, as the `RouteOptions` object is unfortunately optional.

Reply:
Actually, it was a separate error that needed the better error message
https://github.com/mapbox/mapbox-navigation-android/blob/5d022f09807393210b809a579d6e6700315d98d4/libnavigation-core/src/test/java/com/mapbox/navigation/core/replay/route/ReplayRouteMapperTest.kt#L88

But I did just test this with GEOMETRY_POLYLINE and it's an null island, blue screen experience..

How were you replaying a route without `RouteOptions`?

Reply:
> How were you replaying a route without RouteOptions

I requested a route from the API manually, not with the SDK, and then I'm calling `DirectionsRoute#fromJson`. This way, the route will not contain the options. Options are only appended on the SDK (`mapbox-java`) level when making the request.

Reply:
oh i see, since the `RouteAlertsActivity` is doing a lower level approach, i say we change it to use the lower level function and mapGeometry. https://github.com/mapbox/mapbox-navigation-android/pull/3558#discussion_r488728791

that way we can continue giving users a clear error instead of a blue ocean experience when they are mapping mapbox-java requests